conda_build/metadata.py

```diff
@@ -609,6 +612,7 @@ def get_output_dicts_from_metadata(metadata):
     for out in outputs:
         if out.get('name') == metadata.name():
             combine_top_level_metadata_with_output(metadata, out)
+    [trim_empty_keys(out) for out in outputs]
```

Comment:
This could use some future work.  Outputs are coming up with None values for some fields.  That trips up later things, and makes checking values more annoying.  This is a band-aid.  The right fix is to fix the creation of those None values.